lib/charms/layer/jenkins/api.py

```diff
@@ -52,11 +52,7 @@ def __init__(self, packages=None):
 
     @property
     def url(self):
-        config = hookenv.config()
-        prefix = urlparse(config["public-url"]).path
-        if len(prefix) > 0 and prefix[-1] != '/':
-            prefix += '/'
-        return urljoin("http://localhost:8080/", prefix)
+        return "http://localhost:8080/"
```

Comment:
I'm not sure this is what we want. Currently jenkins can be configured to listen on a sub-path of a domain. Won't this mean that no longer works (per your test updates that sound like it's the case as well)?

Reply:
When that prefix is set, Jenkins will still listen in localhost:8080. This means that prior to this change, when setting an URL with a prefix, the configuration change breaks as the charm is unable to reach the API resulting in a 404 error

Reply:
As discussed IRL this seems to work when you deploy with config set, but not when you try to change it afterwards. There'll need to be some code updates to this PR to reflect that.